internal/addrs/removable.go

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) The OpenTofu Authors
+// SPDX-License-Identifier: MPL-2.0
+// Copyright (c) 2023 HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package addrs
+
+// ConfigRemovable is an interface implemented by address types that represents
+// the destination of a "removed" statement in configuration.
+//
+// Note that ConfigRemovable might represent:
+//  1. An absolute address relative to the root of the configuration.
+//  2. A direct representation of these in configuration where the author gives an
+//     address relative to the current module where the address is defined.
+type ConfigRemovable interface {
+	Targetable
+	configRemovableSigil()
```

Comment:
What's the reason behind adding a non-exported member in an exported interface? I'm personally not a fan of exported interfaces with non-exported members as it makes it impossible to implement this interface outside of this package, in which case an interface may not be needed.